test/repository_test.rb

```diff
@@ -115,6 +115,28 @@
           UserRepository.create(user1)
           user1.id.must_equal id
         end
+
+        describe 'when entity is not persisted' do
+          before do
+            @unpersisted_user = User.new(name: 'M', age: '23')
+          end
+          it 'assigns and persists created_at attribute' do
+            result = UserRepository.create(@unpersisted_user)
+            result.created_at.must_equal result.created_at
+          end
+        end
+
+        describe 'when entity is already persisted' do
+          before do
+            user              = User.new(name: 'D', age: '25')
+            @user             = UserRepository.create(user)
+            @persisted_user   = UserRepository.persist(@user)
```

Comment:
*1) Why do you use create then persist here?*

Firstly, I use `.create` to create a User.
After that, I use `.persist` to save User created into DB. 
At the moment, `created_at` will have real value.

*2) Should you only use 1 instead?*
```
before do
  @persisted_user = UserRepository.persist(User.new(name: 'D', age: '25'))
end
```
The way I see that, `User.new(name: 'D', age: '25')` just init a User
After that, you invoke persist method `UserRepository.persist(User.new(name: 'D', age: '25')) `  `created_at` will get value is `nil`. It uncorrect with case `created_at` persisted.